From d9fd9482e34af409a2c79bf91ae05be01e0f549b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 20 Jun 2024 23:18:13 +0800
Subject: [PATCH 03/25] fix(meson.build): more Gallium, Vulkan drivers for on
 all architectures

- This is a pre-upstream patch, I will split this into multiple ones to
  justify each.
- Enable svga Gallium driver by default on ARM targets as VMware can run ARM
  guest machines.
- Enable etnaviv Gallium driver by default on all architectures to support
  older JEMO JM-series GPUs.
- The xe kernel driver is now available and possible to be utilised on all
  known architectures listed in meson.build. Enable iris Gallium drivers
  on all architectures by default.
- Enable more Vulkan drivers following the same spirit as above.
---
 meson.build | 27 +++++++++++++++++++--------
 1 file changed, 19 insertions(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 6b84fe69d5d..1f676a4f3fc 100644
--- a/meson.build
+++ b/meson.build
@@ -144,21 +144,23 @@ if gallium_drivers.contains('auto')
     if ['x86', 'x86_64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'svga', 'swrast',
-        'iris', 'crocus', 'i915', 'zink'
+        'iris', 'crocus', 'i915', 'zink', 'etnaviv'
       ]
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'v3d', 'vc4', 'freedreno', 'etnaviv', 'nouveau', 'svga',
         'tegra', 'virgl', 'lima', 'panfrost', 'swrast', 'iris',
-        'zink'
+        'zink', 'r300', 'r600', 'radeonsi', 'svga', 'asahi'
       ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
       gallium_drivers = [
-        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'swrast', 'zink'
+        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'swrast', 'iris',
+        'zink', 'etnaviv'
       ]
     elif ['loongarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
-        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'etnaviv', 'swrast', 'zink'
+        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'etnaviv', 'swrast',
+        'iris', 'zink'
       ]
     else
       error('Unknown architecture @0@. Please pass -Dgallium-drivers to set driver options. Patches gladly accepted to fix this.'.format(
@@ -221,13 +223,22 @@ _vulkan_drivers = get_option('vulkan-drivers')
 if _vulkan_drivers.contains('auto')
   if system_has_kms_drm
     if host_machine.cpu_family().startswith('x86')
-      _vulkan_drivers = ['amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast', 'virtio'
+      ]
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['swrast', 'intel']
+      _vulkan_drivers = [
+        'amd', 'broadcom', 'freedreno', 'intel', 'nouveau', 'panfrost',
+        'swrast', 'virtio'
+      ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'nouveau', 'swrast', 'virtio'
+      ]
     elif ['loongarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'nouveau', 'swrast', 'virtio'
+      ]
     else
       error('Unknown architecture @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
             host_machine.cpu_family()))
-- 
2.45.2

